### ISSUE TITLE
[Suggestion]: autobuy after vehicle collision / loss

### ISSUE THREAD

**Issue by CactusAcolyte** [JGRennison/OpenTTD-patches#796]
Three vehicle classes as far as I know suffer from crashes:

1. Planes (random crashes + semi-random crashes of larger planes on smaller runways)
2. Road vehicles (semi-random crashes at level crossings)
3. Trains (non-random signalling issues or runaway on steep hills)

With the new longer calendar time scaling patch, the long-running games I've been hosting have my plane-wielding friends suffering from slowly losing their fleet. An auto-buy after loss option would do wonders to make having a plane fleet be sustainable without it having a half-life. Also would be nice to rule out needing to have road bridges over *every* train track out of fear of long-term economic loss.

**Comment by Gadg8eer** [comment]
Agreed on both. Would also be nice to even have "insurance claim" that pays out from another player's pocket if their train destroys your road vehicle.

**Comment by CactusAcolyte** [comment]
> Agreed on both. Would also be nice to even have "insurance claim" that pays out from another player's pocket if their train destroys your road vehicle [...]

maybe that would work better as a gamescript or newgrf? i didn't envision "possible insurance fraud" within the scope of this suggestion
